source/common/http/http2/codec_impl.cc

```diff
@@ -816,7 +816,15 @@ Status ConnectionImpl::onBeforeFrameReceived(const nghttp2_frame_hd* hd) {
          connection_.state() == Network::Connection::State::Open);
 
   current_stream_id_ = hd->stream_id;
-
+  StreamImpl* stream = getStream(hd->stream_id);
```

Comment:
We need to verify that the onBeforeFrameReceived is called after the onBeginHeaders where the stream object is created. I think it may be not. In this case we may want to count frame bytes in the onFrameReceived, which is called after onBeginHeaders

Reply:
Makes sense. Moving it to onFrameReceived do change some count result.